pkg/operator/controller/nodeobservability/scc.go

```diff
@@ -66,39 +71,42 @@ func (r *NodeObservabilityReconciler) desiredSecurityContextConstraints(nodeObs
 	var priority int32 = 10
 
 	scc := &securityv1.SecurityContextConstraints{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: sccName,
-		},
+		ObjectMeta:               metav1.ObjectMeta{Name: sccName},
+		Priority:                 &priority,
 		AllowPrivilegedContainer: true,
-		AllowHostIPC:             false,
+		DefaultAddCapabilities:   nil,
+		RequiredDropCapabilities: []corev1.Capability{"MKNOD"},
+		AllowedCapabilities:      nil,
+		AllowHostDirVolumePlugin: true,
+		Volumes:                  []securityv1.FSType{securityv1.FSTypeHostPath, securityv1.FSTypeSecret, securityv1.FSTypeConfigMap},
 		AllowHostNetwork:         true,
-		AllowHostPID:             false,
 		AllowHostPorts:           false,
-		// This allows us to mount the hosts /var/run/crio/crio.sock into the container
-		AllowHostDirVolumePlugin: true,
-		AllowedCapabilities:      nil,
-		DefaultAddCapabilities:   nil,
-		FSGroup: securityv1.FSGroupStrategyOptions{
-			Type: securityv1.FSGroupStrategyRunAsAny,
-		},
-		Groups:                   []string{"system:cluster-admins", "system:nodes"},
-		Priority:                 &priority,
+		AllowHostPID:             false,
+		AllowHostIPC:             false,
+		SELinuxContext:           securityv1.SELinuxContextStrategyOptions{Type: securityv1.SELinuxStrategyMustRunAs},
+		RunAsUser:                securityv1.RunAsUserStrategyOptions{Type: securityv1.RunAsUserStrategyRunAsAny},
+		SupplementalGroups:       securityv1.SupplementalGroupsStrategyOptions{Type: securityv1.SupplementalGroupsStrategyRunAsAny},
+		FSGroup:                  securityv1.FSGroupStrategyOptions{Type: securityv1.FSGroupStrategyRunAsAny},
 		ReadOnlyRootFilesystem:   false,
-		RequiredDropCapabilities: []corev1.Capability{"MKNOD"},
-		RunAsUser: securityv1.RunAsUserStrategyOptions{
-			Type: securityv1.RunAsUserStrategyRunAsAny,
-		},
-		SELinuxContext: securityv1.SELinuxContextStrategyOptions{
-			Type: securityv1.SELinuxStrategyMustRunAs,
-		},
-		SupplementalGroups: securityv1.SupplementalGroupsStrategyOptions{
-			Type: securityv1.SupplementalGroupsStrategyRunAsAny,
-		},
-		Volumes: []securityv1.FSType{securityv1.FSTypeHostPath, securityv1.FSTypeSecret, securityv1.FSTypeConfigMap},
+		Groups:                   []string{"system:cluster-admins", "system:nodes"},
 	}
 	return scc
 }
 
+func (r *NodeObservabilityReconciler) updateSecurityContextConstraintes(ctx context.Context, current, desired *securityv1.SecurityContextConstraints) (bool, error) {
+	opts := cmpopts.IgnoreFields(securityv1.SecurityContextConstraints{}, "TypeMeta", "Users", "SeccompProfiles", "AllowedUnsafeSysctls", "ForbiddenSysctls")
```

Comment:
Why these are filtered out: `"SeccompProfiles", "AllowedUnsafeSysctls", "ForbiddenSysctls"` ?

Reply:
These fields were previously unset and looks like we are just relying on the defaults. Do you want me to explicitly add them to desired? (at least for the ones that have values as defaults)

Reply:
I think that relying on the defaults makes sense for "workload resources" (PODs, deployments, etc.). SCC is a sort of security framework which will put boundaries on what the agent can do. I think we should be precise in every field of this framework.
If we don't set some fields than they should remain empty because otherwise the customer can update them: add allowed unsafe/forbidded sysctls, etc. We should ask this question to ourselves: do we want the agent to be added with unsafe/forbidden syscalls? I'd say no unless there is a strict necessity.
Although `Users` and `Groups`, I'd keep them ignored to let the customers to add other SAs to this SCC.